echopype/convert/set_groups_azfp.py

```diff
@@ -167,6 +168,7 @@ def set_platform(self) -> xr.Dataset:
         """Set the Platform group."""
         platform_dict = {"platform_name": "", "platform_type": "", "platform_code_ICES": ""}
         unpacked_data = self.parser_obj.unpacked_data
+        depth = depth_from_pressure(unpacked_data["pressure"], latitude=30, atm_pres_surf=0)
```

Comment:
pressure will rarely be available in AZFP data. So, here, there needs to be a handler to check if `"pressure" in unpacked_data` (I can't remember, but I assume that if there was no pressure data present, `unpacked_data` doesn't contain a pressure entry?). If pressure is not present, set `depth = [np.nan] * len(time2)`, as was done before. Except `time2` hasn't been declared here yet, so you should move this `depth` block to just before the `ds = xr.Dataset` statement.

Reply:
`pressure` data will always be available in the parsed data and if in case the required parameters for calculating `pressure` aren't available, then the value be set to `np.nan`.So, I don't think there will be a case with `pressure` data not being present.

Reply:
Ok. I think I was unclear, but I understand what you mean. `unpacked_data` will always have a `pressure` entry, but it may be set to `np.nan`.

In that case, though, it seems like an unnecessary step to apply the function `depth_from_pressure` if we know in advance that pressure is np.nan. So, how about something like:
```suggestion
        if unpacked_data["pressure"] == np.nan:
            depth = np.nan
        else:
            depth = depth_from_pressure(unpacked_data["pressure"], latitude=30, atm_pres_surf=0)
```

Also, as a side note, I need to ask others about pressure measurements and whether it's typical that `atm_pres_surf` (atmospheric pressure) is already removed.